src/components/YourCompany/BusinessContactInfo.vue

```diff
@@ -1,293 +1,77 @@
-  }
-
-  private get hasBusinessContactInfoChange () {
-    return !this.isSame(this.businessContact, this.getOriginalBusinessContact())
-  }
+import { AuthApiMixin, CommonMixin } from '@/mixins'
 
-  private updateContactInfo (): void {
-    this.emitContactInfo(this.contact)
-    this.isEditing = false
-    this.emitHaveChanges(this.hasBusinessContactInfoChange)
-  }
+// Components
+import { ContactInfo } from '@bcrs-shared-components/contact-info'
 
-  private resetContactInfo (): void {
-    this.contact = this.getOriginalBusinessContact()
-    this.emitContactInfo(this.contact)
-    this.emitHaveChanges(false)
-    this.isEditing = false
+@Component({
+  components: {
+    ContactInfo
   }
-
-  private getOriginalBusinessContact (): BusinessContactIF {
-    return {
-      email: this.originalBusinessContact.email,
-      confirmEmail: this.originalBusinessContact.email,
-      phone: this.originalBusinessContact.phone,
-      extension: this.originalBusinessContact.extension
-    }
+})
+export default class BusinessContactInfo extends Mixins(AuthApiMixin, CommonMixin) {
+  @Action setBusinessContact!: ActionBindingIF
+
+  @Getter getBusinessContact!: BusinessContactIF
+  @Getter getOriginalIA!: IncorporationFilingIF
+  @Getter getOriginalSnapshot!: BusinessSnapshotIF
+
+  /** Get the original Contact info dependant on filing type. */
+  private get originalContact (): BusinessContactIF {
+    return this.isCorrectionView()
+      ? this.getOriginalIA.incorporationApplication.contactPoint
+      : this.getOriginalSnapshot[5]
   }
 
-  private get hasPhoneNumberChange (): boolean {
-    return this.businessContact.phone !== this.originalBusinessContact.phone ||
-    this.businessContact.extension !== this.originalBusinessContact.extension
+  /** Check for changes between current contact and original contact. */
+  private get hasBusinessContactInfoChange (): boolean {
+    return this.getBusinessContact?.email !== this.originalContact?.email ||
+      this.getBusinessContact?.phone !== this.originalContact?.phone ||
+      this.getBusinessContact?.extension !== this.originalContact?.extension
   }
 
-  private get hasEmailAddressChange (): boolean {
-    return this.businessContact.email !== this.originalBusinessContact.email
+  /** Update Contact info. */
+  private async setContact (contactInfo: BusinessContactIF): Promise<void> {
+    this.isCorrectionView()
+      ? this.setBusinessContact(contactInfo)
+      : await this.updateContactRequest(contactInfo)
   }
 
-  // Watchers
-  @Watch('businessContact', { deep: true, immediate: true })
-  private onContactPropValueChanged (): void {
-    this.contact = { ...this.businessContact }
-    this.emitHaveChanges(this.hasBusinessContactInfoChange)
+  /** Request to update contact info in Auth and Store. */
+  private async updateContactRequest (contactInfo: BusinessContactIF): Promise<void> {
+    await this.updateContactInfo(contactInfo)
+    this.setBusinessContact(contactInfo)
```

Comment:
Should we await store setters? In NR some effort was made to handle the supposedly async store setters but in our other UIs we've never worried about this and I don't think we've seen any problems. Thoughts?

Reply:
The asynchronous call isn't setting anything to store itself, it is making the request to update the information in Auth. 
I chose to set to store after that request in the event it fails, so we don't have data in the UI that is out of sync (behind) what we have in Auth.

Reply:
No, the line I commented on
`this.setBusinessContact(contactInfo)`
is a store setter, right?

Reply:
Yes, `this.setBusinessContact(contactInfo)` is a store setter. 
You think this could be an issue, setting after waiting for a successful request?

Reply:
I think the NR guys were concerned enough about store setters that they awaited those calls.

Meanwhile, we never have and we seem to be OK.

So I dunno. Is it an issue? Is it not an issue? 🤷‍♂️

Reply:
Yea, ditto, we've never had an issue. Could be the way we've architected the setters (actions), and bound them. 
This isn't a problem i don't think.

Reply:
They `awaited` ALL their store setters? I don't recall this.. but that certainly seems fishy. We are just defining a key's value essentially, what would we need to wait for 🤔 

Perhaps, since they had a lot of sequential coding in the modules which would dictate next steps that would require those aforementioned values to be present, they had to ensure these values set before proceeded? 

Sounds like a UI not being in sync with the store kinda nightmare lol

Reply:
Sounds like what we've been seeing in the Sentry logs.

OK, no change neede here.